ostd/src/mm/page/mod.rs

```diff
@@ -83,6 +83,30 @@ impl<M: PageMeta> Page<M> {
         }
     }
 
+    /// Check whether the page described by paddr is allocated.
+    /// true if the page is allocated, false otherwise.
+    /// 
+    /// # Panics
+    /// 
+    /// The function panics if:
+    /// 
+    /// - the physical address is out of bound or not aligned.
+    pub fn check_page_status(paddr: Paddr) -> bool {
```

Comment:
A way to ensure that the status of the page will not be changed after the load would be marking this function unsafe. The caller should ensure that no other threads will do `from_unused` or `drop`. So it is better to confine the `from_unused` calls into a single module with a given set of APIs, like in the old way of `page::allocator`, with a few exceptions. The caller should take a big lock on the global page allocator too.